src/Payments.Mvc/Controllers/FinancialAccountsController.cs

```diff
@@ -0,0 +1,493 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Payments.Core.Data;
+using Payments.Core.Domain;
+using Payments.Core.Extensions;
+using Payments.Mvc.Identity;
+using Payments.Mvc.Models.FinancialModels;
+using Payments.Mvc.Models.Roles;
+using Payments.Mvc.Services;
+
+namespace Payments.Mvc.Controllers
+{
+    public class FinancialAccountsController : SuperController
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IFinancialService _financialService;
+
+        public FinancialAccountsController(ApplicationDbContext context, IFinancialService financialService, ApplicationUserManager userManager)
+            : base(userManager)
+        {
+            _context = context;
+            _financialService = financialService;
+        }
+
+        /// <summary>
+        /// GET: FinancialAccounts/Create
+        /// </summary>
+        /// <param name="id">Team id</param>
+        /// <returns></returns>
+        public async Task<IActionResult> CreateAccount(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var team = await _context.Teams
+                .SingleOrDefaultAsync(m => m.Id == id && m.IsActive);
+            if (team == null)
+            {
+                return NotFound();
+            }
+            if (!User.IsInRole(ApplicationRoleCodes.Admin) && !await _context.TeamPermissions.AnyAsync(a => a.TeamId == team.Id && a.UserId == CurrentUserId))

```

Comment:
@jSylvestre definitely will want to move this to a policy or somehow out of the controller eventually